pyomo/gdp/plugins/chull.py

```diff
@@ -210,231 +233,205 @@ def _apply_to_impl(self, instance, **kwds):
+        if len(obj.disjuncts) == 0:
+            raise GDP_Error("Disjunction %s is empty. This is " 
+                            "likely indicative of a modeling error."  %
+                            obj.getname(fully_qualified=True,
+                                        name_buffer=NAME_BUFFER))
 
         # We first go through and collect all the variables that we
         # are going to disaggregate.
         varOrder_set = ComponentSet()
         varOrder = []
         varsByDisjunct = ComponentMap()
+        localVarsByDisjunct = ComponentMap()
         for disjunct in obj.disjuncts:
-            # This is crazy, but if the disjunct has been previously
-            # relaxed, the disjunct *could* be deactivated.
-            not_active = not disjunct.active
-            if not_active:
-                disjunct._activate_without_unfixing_indicator()
-            try:
-                disjunctVars = varsByDisjunct[disjunct] = ComponentSet()
-                for cons in disjunct.component_data_objects(
-                        Constraint,
-                        active = True,
-                        sort=SortComponents.deterministic,
-                        descend_into=Block):
-                    # we aren't going to disaggregate fixed
-                    # variables. This means there is trouble if they are
-                    # unfixed later...
-                    for var in EXPR.identify_variables(
-                            cons.body, include_fixed=False):
-                        # Note the use of a list so that we will
-                        # eventually disaggregate the vars in a
-                        # deterministic order (the order that we found
-                        # them)
-                        disjunctVars.add(var)
-                        if var not in varOrder_set:
-                            varOrder.append(var)
-                            varOrder_set.add(var)
-            finally:
-                if not_active:
-                    disjunct._deactivate_without_fixing_indicator()
-
-        # We will only disaggregate variables that
-        #  1) appear in multiple disjuncts, or
-        #  2) are not contained in this disjunct, or
-        #  3) are not themselves disaggregated variables
+            disjunctVars = varsByDisjunct[disjunct] = ComponentSet()
+            for cons in disjunct.component_data_objects(
+                    Constraint,
+                    active = True,
+                    sort=SortComponents.deterministic,
+                    descend_into=Block):
+                # [ESJ 02/14/2020] We *will* disaggregate fixed variables on the
+                # philosophy that fixing is not a promise for the future and we
+                # are mathematically wrong if we don't transform these correctly
+                # and someone later unfixes them and keeps playing with their
+                # transformed model
```

Comment:
Oh... Yikes. The answer *is* no (even in the linear case). And I'm of two minds about it: Because if you are striving for the tightest-possible M values, then it would be very annoying if fixed variables were treated as unfixed. But it's definitely counter to the above philosophy. Perhaps the user should have a choice about how to treat fixed variables in the transformation?

Reply:
Having an option for the transformation makes a lot of sense.